filebeat/module/nats/log/test/test.log-expected.json

```diff
@@ -73,7 +76,12 @@
         "input.type": "log",
         "log.level": "trace",
         "log.offset": 359,
-        "message": "172.18.0.1:38630 - cid:1 - ->> [CONNECT {\"verbose\":false,\"pedantic\":false,\"tls_required\":false,\"name\":\"NATS Benchmark\",\"lang\":\"go\",\"version\":\"1.7.0\",\"protocol\":1,\"echo\":true}]",
+        "nats.log.client.id": "1",
+        "nats.log.client.ip": "172.18.0.1",
+        "nats.log.client.port": "38630",
+        "nats.log.msg.data": "{\"verbose\":false,\"pedantic\":false,\"tls_required\":false,\"name\":\"NATS Benchmark\",\"lang\":\"go\",\"version\":\"1.7.0\",\"protocol\":1,\"echo\":true}",
```

Comment:
@sayden This field always contains a json string. Do we need to manipulate it further or leave it as is? If yes, how can we do it?

Reply:
Frankly, I haven't faced this situation yet, sorry. You can use https://www.elastic.co/guide/en/elasticsearch/reference/current/json-processor.html in the pipeline to explode that JSON. The key thing here is that those fields must be constant between different NATS installations and I guess they aren't. I mean that if you explode it and convert them to fields, they'll need to be indexed unless they are not relevant. @jsoriano do you have any thoughts here?

Reply:
I agree with you @sayden. It depends if the fields contained in the JSON are relevant, and if they are a known set fields.

If you only want to extract some of the fields you could try to:
* Parse the json to a temporal field
* Rename the fields you want to collect
* Remove the temporal field with the rest of fields

You could still keep the raw data if you want.

Reply:
@sayden Actually both **CONNECT** and **INFO** type messages contain specific fields inside their dictionaries according to NATS documentation [here](https://nats.io/documentation/internals/nats-protocol/#a-name-connect-a-connect).

The documentation does not mention differences between versions, but still it is documentation and can vary from reality, I am ok to leave the field as is, if all of us (committers and reviewers) agree.

Thank you for pointing the json processor out. The only question remains is if we are going to use it or not. What is the others' opinion?

Reply:
Hey, I only see `clientd_id` of INFO, and `user` of CONNECT fields to be interesting. However, I think that this info(as @skatsaounis reminded me), the CID, is already scrapped on a previous step of the pipeline. So, realistically speaking I don't see any reason on adding things. Correct me if I am mistaken.

Reply:
I also believe that these fields do not add any value for a NATS user. So it's ok not to process them any further.